File: BE-Task_1-Java/src/main/java/io/mekari/assessment/refundbackendjava/repository/AgentRepository.java
package io.mekari.assessment.refundbackendjava.repository;

import io.mekari.assessment.refundbackendjava.domain.model.Agent;
import java.util.List;
import java.util.Optional;

public interface AgentRepository {
    Optional<Agent> findById(String agentId);

    List<Agent> listAll();
}

File: BE-Task_1-Php/internal/repository/interfaces/agent_repository.php
<?php

declare(strict_types=1);

namespace App\Internal\Repository\Interfaces;

use App\Internal\Domain\Entities\Agent;

interface AgentRepositoryInterface
{
    public function findById(string $agentId): ?Agent;

    /**
     * @return list<Agent>
     */
    public function listAll(): array;
}
